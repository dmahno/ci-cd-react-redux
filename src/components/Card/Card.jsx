import React from 'react';
import './Card.scss';
import CardTitle from '../CardTitle/CardTitle';
import CardUserMeta from '../CardUserMeta/CardUserMeta';
import CardTimeMeta from '../CardTimeMeta/CardTimeMeta';
import './Card.scss';
import CardSatus from '../CardStatus/CardSatus';
import { Link } from 'react-router-dom';

function Card({
  cardStatus,
  colorNumber,
  titleNumber,
  commitNumber,
  brunchName,
  userName,
  date,
  time,
  buildId,
  cardTitle
}) {
  return (
    <Link to={`/details?buildId=${buildId}`}>
      <div className='card'>
      <div className='card__data'>
        <CardSatus cardStatus={cardStatus} />
        <div className='card__content-meta'>
          <div className='card__title-user-meta'>
            <CardTitle
              cardTitle={cardTitle}
              colorNumber={colorNumber}
              titleNumber={titleNumber}
            />
            <CardUserMeta
              brunchName={brunchName}
              commitNumber={commitNumber}
              userName={userName}
            />
          </div>
          <div className='card__time-meta'>
            <CardTimeMeta date={date} time={time} />
          </div>
        </div>
      </div>
    </div>
    </Link>
  );
}

export default Card;
