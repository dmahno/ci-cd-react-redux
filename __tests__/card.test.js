import renderer from "react-test-renderer";
import React from "react";
import Card from "../src/components/Card/Card";
import { MemoryRouter } from "react-router-dom";

describe("Card component renders contents correctly", () => {
  it("renders correctly", () => {
    const cardData = {
      brunchName: "master",
      buildId: "5e79a76e-13a2-472d-8aa5-19d52c3e33a1",
      cardStatus: "clock",
      cardTitle: "FP-2020: INITIAL↵",
      colorNumber: "pending",
      commitNumber: "573865",
      date: "21 янв, 03:06",
      time: "1 ч 20 мин",
      userName: "akimy",
      titleNumber: 1
    };
    const rendered = renderer.create(
      <MemoryRouter>
        <Card {...cardData} />
      </MemoryRouter>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
