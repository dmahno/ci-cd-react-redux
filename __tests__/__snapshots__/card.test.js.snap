// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders contents correctly renders correctly 1`] = `
<a
  href="/details?buildId=5e79a76e-13a2-472d-8aa5-19d52c3e33a1"
  onClick={[Function]}
>
  <div
    className="card"
  >
    <div
      className="card__data"
    >
      <div
        className="card-status"
      >
        <i
          className="icon clock"
        />
      </div>
      <div
        className="card__content-meta"
      >
        <div
          className="card__title-user-meta"
        >
          <div
            className="card-title"
          >
            <span
              className="card-title__number pending"
            >
              #
              1
            </span>
            <span
              className="card-title__text"
            >
              FP-2020: INITIAL↵
            </span>
          </div>
          <div
            className="user-meta"
          >
            <div
              className="user-meta__branches"
            >
              <i
                className="icon code-commit"
              />
              <div
                className="user-meta__branches-name"
              >
                master
              </div>
              <div
                className="user-meta__commit-number"
              >
                573865
              </div>
            </div>
            <div
              className="user-meta__user-info"
            >
              <i
                className="icon user"
              />
              <div
                className="user-meta__user-info-name"
              >
                akimy
              </div>
            </div>
          </div>
        </div>
        <div
          className="card__time-meta"
        >
          <div
            className="time-meta padding_top_8_mobile_768 flex_row_mobile_768"
          >
            <div
              className="time-meta__date"
            >
              <i
                className="icon calendar"
              />
              21 янв, 03:06
            </div>
            <div
              className="time-meta__time margin_top_12 margin_mobile_reset_768 "
            >
              <i
                className="icon stopwatch"
              />
              1 ч 20 мин
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</a>
`;
